### ISSUE TITLE
Confidence scores for predictions

### ISSUE THREAD

**Issue by mlhoggard** [KennthShang/PhaGCN#4]
Hi there,

After trying out HostG (thanks for you help and responses with that!) I spotted this and thought it'd be great to try out on the same data.

Similar to my question with HostG, I was wondering if the taxonomy predictions have any sort of confidence score associated with them, and if it was possible to output this with `final_prediction.csv` at all? 

Currently, I can't tell if the predictions are confident ones, or simply the "closest" match (but may have very low confidence for the match?). I was also wondering if different taxonomic ranks had different confidence scores associated with them (again, similar to our conversation in the HostG thread - thanks again for implementing that!).

Also, just a quick side note that I think `MCL` should also to be added to the dependency list in `README.md`? I had a failed attempt the first time and eventually noticed it failed from not having something from MCL loaded at the time...

Kind regards,
Mike.

**Comment by KennthShang** [comment]
The softmax value will also be outputted with the prediction. 

I also updated the MCL description in README.md. Thanks.


Best,
Jiayu

**Comment by mlhoggard** [comment]
Hi Jiayu,

Thanks for the swift reply and update! I gave the updated version a test and all works well thank you.

Just as a quick follow-up: 

Do you have an indication of appropriate softmax value thresholds whereby you can be reasonably confident in the taxonomy assignment? 

I.e. Prior to this update, were the output taxonomy predictions filtered for those over *x* softmax value, or does the program simply output the closest taxonomy match based on the model constructed from the training set? 

And if the latter, what would the expected behaviour be for non-*Caudovirales* viruses? Would these be returned as "unclassified", or with a *Caudovirales* family classification but with low softmax value?

Thanks again,
Mike.

**Comment by KennthShang** [comment]
Actually, there is no further experiment to show the softmax threshold for PhaGCN. 

But the database will be updated soon. Users can also update the database according to their needs. The 2.0 version will cover the taxonomy prediction for all the viruses.

Best,
Jiayu

**Comment by mlhoggard** [comment]
Hi Jiayu,

Thanks for the reply and info. Looking forward to the 2.0 update - I'll keep an eye out for it!

Cheers,
Mike.
